app/models/topic.rb

```diff
@@ -1384,10 +1384,15 @@ def remove_banner!(user)
     Jobs.cancel_scheduled_job(:remove_banner, topic_id: self.id)
   end
 
-  def banner
+  def banner(guardian = nil)
     post = self.ordered_posts.first
 
-    { html: post.cooked, key: self.id, url: self.url }
+    html = post.cooked
+    if (guardian && ContentLocalization.show_translated_post?(post, guardian))
+      html = post.get_localization&.cooked.presence || html
+    end
```

Comment:
```suggestion
    html = 
      if (guardian && ContentLocalization.show_translated_post?(post, guardian))
        post.get_localization&.cooked.presence
      end

    html ||= post.cooked
```

Little bit of code golf.